modules/core/entity/farm_entity.module

```diff
@@ -68,7 +68,7 @@ function farm_entity_entity_type_build(array &$entity_types) {
   }
 
   // Enable the use of bundle plugins on specific entity types.
-  foreach (['asset', 'log', 'plan', 'plan_record', 'quantity'] as $entity_type) {
+  foreach (['asset', 'data_stream', 'log', 'plan', 'plan_record', 'quantity'] as $entity_type) {
```

Comment:
If we make this change I think we also need to include `data_streams` in the logic of `farm_entity_entity_field_storage_info_alter()`. Otherwise contributed bundle fields will not have the correct module provider and can cause issues when installing/uninstalling the modules that provide these bundle fields. It seems that `plan_record` might already be missing from this logic.

I think we could also remove the `data_stream` module dependency on `entity`, and add a dependency on `farm_entity` instead UNLESS we move the `DataStreamTypeManager` and etc `DataStreamType` plugin scaffolding to the `farm_entity` module like we do for our other entity types.